web/src/channel.js

```diff
@@ -201,3 +201,18 @@ export function xhr_error_message(message, xhr) {
     }
     return message;
 }
+
+export async function async_request(request_method, url, data) {
+    return await new Promise((resolve) => {
+        request_method({
+            url,
+            data,
+            success(data, textStatus, jqXHR) {
+                resolve({data, textStatus, jqXHR});
+            },
+            error(xhr, error_type, xhn) {
+                resolve({xhr, error_type, xhn});
```

Comment:
Not rejecting because Promise.all will executing other requests after a request fails in middle.